src/components/Search/index.tsx

```diff
@@ -58,7 +58,7 @@ function mapTransactionItemToSelectedEntry(item: TransactionListItemType): [stri
         item.keyForList,
         {
             isSelected: true,
-            canDelete: item.canDelete,
+            canDelete: canDeleteCardTransactionByLiabilityType(item.transactionID, item.canDelete),
```

Comment:
What if the full transaction object is not present in onyx? the Search call does not return all the transaction objects, does it? I think we need to add it to the Search response, what do you see in your testing @narefyev91 @allgandalf ?

Reply:
Seems like I already have it there as we return the full comment JSON so we are good 
<img width="331" alt="image" src="https://github.com/user-attachments/assets/915d0993-0155-4174-a838-134e7809e971" />

Reply:
@narefyev91 I wonder if we should not use the `canDelete` param. Does it not hold the correct value?

Reply:
I can look into fixing the logic in BE

Reply:
Created a backend fix that should ensure the canDelete returns correctly false or true based on the card liability type. Then I guess we wont need this  PR, right? https://github.com/Expensify/Auth/pull/14746 as I assume the logic already looks are the canDelete flag

Reply:
Did we?